### PR TITLE
fix #14909 issue

### LINKED ISSUE #14909: PageView::addEventListener() is invalid in v3.10
m_PageView->addEventListener(TestScene::callbackFunc, this);
TestScene::callbackFunc don't called when m_PageView turning.
sorry my English is poor.

### PR DESCRIPTION
**Pull request by Naruto**
This PR adds autoscroll finish event to ui::ScrollView , and changes PageView::addEventListener parameter type from PageView::EventType to ScrollView::EventType. 

PageView event is not invoked at auto scroll fishing since v3.10. Because PageView inherited ScrollView.

So, I added autoscroll finish event ot ui::ScrollView, added ScreenView::addEventListener to PageView class.

please review this PR.

### CODE REVIEW

**Comment by zilongshanren** [comment]
@Naruto 
I'm OK with this change, but why deprecated the original `addEventListener` of PageView?

@neokim  please help to review it thanks.

File: cocos/ui/UIPageView.h [hunk 1]
```diff
@@ -25,6 +25,7 @@ THE SOFTWARE.
 #ifndef __UIPAGEVIEW_H__
 #define __UIPAGEVIEW_H__
 
+#include "ui/UIScrollView.h"
```
Comment (zilongshanren) [inline cocos/ui/UIPageView.h:28]: ui/UIScrollView is already included in UIListView, this might be useless.

**Comment by zilongshanren** [comment]
PageView emits a different event at page turning.

``` cpp
  enum class EventType
  {         
         TURNING
   }
```

**Comment by zilongshanren** [comment]
https://github.com/cocos2d/cocos2d-x/issues/14909

**Comment by Naruto** [comment]
@zilongshanren  emitting TURNING is not implemented now. so I added deprecated.

**Comment by zilongshanren** [comment]
@Naruto 
New PR is https://github.com/cocos2d/cocos2d-x/pull/14964, thanks for your idea for fixing this issue.
I added some improvements over your solution, please take a look.

**Comment by Naruto** [comment]
@zilongshanren I see.

**Comment by zilongshanren** [comment]
@Naruto 
I also fixed the binding issues on jsb and lua.

**Comment by Naruto** [comment]
@zilongshanren :+1:

**Pull request closed without merging**